narwhals/_dask/namespace.py

```diff
@@ -51,8 +51,10 @@ def col(self, *column_names: str) -> DaskExpr:
         )
 
     def all_horizontal(self, *exprs: IntoDaskExpr) -> DaskExpr:
-        # coverage shows 55->exit as uncovered?
-        return reduce(lambda x, y: x & y, flatten(exprs))  # type: ignore[no-any-return]  # pragma: no cover
+        return reduce(lambda x, y: x & y, parse_into_exprs(*exprs, namespace=self))
```

Comment:
with `flatten` it would only work by passing `nw.col` (or an expression in general). With parsing we can use string directly. I added tests for that